File: Sliding-Window/1004. Max Consecutive Ones III/longestOnes.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        output = 0
        numT = numF = 0
        left = right = 0
        
        while right < len(nums):
            if nums[right] == 1:
                numT += 1
            else:
                numF += 1
            
            if numF > k:
                if nums[left] == 1:
                    numT -= 1
                else:
                    numF -= 1
                left += 1
            
            output = max(output, right - left + 1)
            right += 1
        
        return output

File: Sliding-Window/1004. Max Consecutive Ones III/longestOnes.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
// count 1
 var longestOnes = function(nums, k) {
    let output = 0;
    let left = right = 0;
    let numT = numF = 0;
    
    // slide right
    while (right < nums.length) {
        if (nums[right] === 1) 
            numT++;
        else
            numF++;
        
        // if numF more than k
        // means we can't flip anymore
        // slide left
        while (numF > k) {
            if (nums[left] === 1) 
                numT--;
            else
                numF--;
            left++;
        }
        
        output = Math.max(output, right - left + 1);
        right++;
    }
    
    return output;
};